src/v/security/audit/audit_log_manager.cc

```diff
@@ -847,6 +889,47 @@ bool audit_log_manager::report_redpanda_app_event(is_started app_started) {
         : application_lifecycle::activity_id::stop);
 }
 
+model::partition_id audit_log_manager::compute_partition_id() {
+    static thread_local model::partition_id _next_pid{0};
+
+    model::topic_namespace_view ns_tp{model::kafka_audit_logging_nt};
+    auto cfg = _metadata_cache->local().get_topic_cfg(ns_tp);
+    if (cfg.has_value()) {
+        vlog(
+          adtlog.debug,
+          "{} missing from metadata cache, fall back to partition 0",
+          ns_tp);
+        return _next_pid;
+    }
+    auto n_partitions = cfg.value().partition_count;
+    vassert(n_partitions >= 0, "Invalid partition count {}", n_partitions);
+
+    auto inc_pid = [n_partitions](model::partition_id pid, int32_t inc = 1) {
+        return model::partition_id{(pid + inc) % n_partitions};
+    };
+
+    const auto& partition_leaders
+      = _controller->get_partition_leaders().local();
+
+    std::optional<model::partition_id> pid;
+    for (auto i : boost::irange(n_partitions)) {
+        auto try_pid = inc_pid(_next_pid, i);
+        auto leader = partition_leaders.get_leader(ns_tp, try_pid);
+        if (!leader.has_value()) {
+            continue;
+        } else if (leader.value() == _self) {
+            vlog(adtlog.debug, "Node {} leads partition {}", _self, try_pid);
+            pid.emplace(try_pid);
+            break;
+        }
+    }
+
+    // NOTE(oren): sort of arbitrary. if we didn't find a locally led partition,
+    // then at least advance the round robin to the next PID in natural order
+    _next_pid = inc_pid(pid.value_or(_next_pid));
+    return pid.value_or(_next_pid);
```

Comment:
praise: took a moment (and some paper & pen) to walk through the logic, but it appears sound to me.  Thank you!